polling_stations/apps/addressbase/management/commands/update_addressbase.py

```diff
@@ -0,0 +1,167 @@
+        self.stdout.write(f"uprntocouncil_path to {uprntocouncil_path}")
+
+        addressbase_updater = AddressbaseUpdater()
+        uprntocouncil_updater = UprnToCouncilUpdater()
+
+        # Set the data_path on each updater instance
+        addressbase_updater.data_path = addressbase_path
+        uprntocouncil_updater.data_path = uprntocouncil_path
+
+        # Get constraints and index information for both tables
+        addressbase_updater.get_constraints_and_index_statements()
+        uprntocouncil_updater.get_constraints_and_index_statements()
+
+        try:
+            # Create empty temp tables for both
+            self.stdout.write("Creating temporary tables...")
+            addressbase_updater.create_temp_table()
+            uprntocouncil_updater.create_temp_table()
+
+            # Import data into temp tables
+            self.stdout.write("Importing data into temporary tables...")
+            addressbase_updater.import_data_to_temp_table()
+            uprntocouncil_updater.import_data_to_temp_table()
+
+            # Add temp primary keys
+            self.stdout.write("Adding primary keys to temporary tables...")
+            addressbase_updater.add_temp_primary_key()
+            uprntocouncil_updater.add_temp_primary_key()
+
+            # Build temp indexes
+            self.stdout.write("Building indexes on temporary tables...")
+            addressbase_updater.build_temp_indexes()
+            uprntocouncil_updater.build_temp_indexes()
+
+            # Perform the table swaps in a single transaction
+            with transaction.atomic():
+                self.stdout.write("Starting atomic transaction for table swaps...")
+
+                # Drop all foreign keys first
+                if addressbase_updater.foreign_key_constraints:
+                    addressbase_updater.drop_foreign_keys()
+                if uprntocouncil_updater.foreign_key_constraints:
+                    uprntocouncil_updater.drop_foreign_keys()
+
+                # Rename old tables
+                addressbase_updater.drop_old_table()
+                uprntocouncil_updater.drop_old_table()
+
+                # Rename temp tables
+                addressbase_updater.rename_temp_table()
+                uprntocouncil_updater.rename_temp_table()
+
+                # Add foreign keys back
+                if addressbase_updater.foreign_key_constraints:
+                    addressbase_updater.add_foreign_keys()
+                if uprntocouncil_updater.foreign_key_constraints:
+                    uprntocouncil_updater.add_foreign_keys()
+
+            self.stdout.write(
+                self.style.SUCCESS(
+                    "Successfully updated both Addressbase and UPRN to Council tables"
+                )
+            )
+
+        finally:
+            # Clean up both updaters
+            self.stdout.write("Cleaning up...")
+            addressbase_updater.db_cleanup()
+            uprntocouncil_updater.db_cleanup()
```

Comment:
If we downloaded the files from S3 and saved them to temp, can we clean up the temp files in the finally block too?